### PR TITLE
get apis

### PR DESCRIPTION
**Pull request by yeudit**
new get ap's for test_suite and test_run models
code cleanup
tests with 100% coverage

includes code from [db/US127906/save-data-in-db](https://github.com/att/qujata/tree/db/US127906/save-data-in-db) :
create database models
create test suite and test runs for every analyze process
some code organization and cleanup

### CODE REVIEW

File: api/src/services/analyze_service.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        'to': end_time
+    })
+
+def __create_test_run(algorithm, iterations, test_suite_id):
+    start_time=datetime.now()
+    __run_test(algorithm, iterations)
+    end_time=datetime.now()
+    tests_service.create_test_run(start_time, end_time, algorithm, iterations, test_suite_id)
+
+def __run_test(algorithm, iterations):
```
Comment (mikekcs) [inline api/src/services/analyze_service.py:44]: change run_test to run

File: api/src/services/analyze_service.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    # end time is now + 90 sec, to show the graph after the test for sure finished running
+    end_time = int(datetime.timestamp(datetime.now() + timedelta(seconds=90)) * 1000)
+
+    return jsonify({
+        'from': start_time,
+        'to': end_time
+    })
+
+def __create_test_run(algorithm, iterations, test_suite_id):
```
Comment (mikekcs) [inline api/src/services/analyze_service.py:38]: Add status
SUCCESS (or similar)
FAILED: if Failed continue to process with others test_run without exception

File: api/src/services/tests_service.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from src.models.test_run import TestRun
+from src.exceptions.exceptions import ApiException
+
+def create_test_suite(data):
+    env_info = current_app.database_manager.get_last_record(EnvInfo)
+    if env_info == None:
+        raise ApiException('Missing env info in database', 'Analyze test failed to complete', 422)
+
+    test_suite = TestSuite(
+        protocol="TLS 1.3",
```
Comment (mikekcs) [inline api/src/services/tests_service.py:20]: move to config

File: api/src/services/tests_service.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return current_app.database_manager.get_records(TestSuite)
+
+def get_test_suite(test_suite_id):
+    return current_app.database_manager.get_record_by_id(TestSuite, test_suite_id)
+
+def get_test_runs(test_suite_id):
+    return current_app.database_manager.get_records(TestRun, [TestRun.test_suite_id == test_suite_id])
+
+def get_test_run(test_suite_id, test_run_id):
+    return current_app.database_manager.get_record_by_id(TestRun, test_run_id)
```
Comment (mikekcs) [inline api/src/services/tests_service.py:57]: test_run_id and test_suite_id

File: mysql/scripts/init-db.sql [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                           FOREIGN KEY (test_suite_id) REFERENCES test_suites(id)
+);
+
+CREATE TABLE IF NOT EXISTS test_run_results (
+                                  test_run_id INT,
+                                  metric_name ENUM('Average CPU', 'Average Memory', 'Error rate', 'Bytes Throughput Per Second', 'Messages Throughput Per Second', 'Average TLS Handshake Time'),
+                                  value BIGINT,
+                                  PRIMARY KEY (test_run_id, metric_name),
+                                  FOREIGN KEY (test_run_id) REFERENCES test_runs(id)
+);
```
Comment (mikekcs) [inline mysql/scripts/init-db.sql:52]: Add default (sample) value

File: api/src/controllers/env_info_api.py
[the remark this answers is not in the document]
Comment (yeudit) [reply api/src/controllers/env_info_api.py:1]: should resolve by @kaylarizi

File: api/src/controllers/env_info_api.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+expected_keys = ['resource_name', 'operating_system', 'cpu', 'cpu_architecture', 'cpu_cores', 'clock_speed', 'node_size']
+
+
+@env_info_api.route('/env-info', methods=['POST'])
+@cross_origin(origins=['*'], supports_credentials=True)
+def insert_env_info():
+    # Extract data from the request
+    data = request.get_json()
+
+    # Check that at least one expected key-value pair exists in the JSON payload
```
[the remark this answers is not in the document]
Comment (yeudit) [reply api/src/controllers/env_info_api.py]: should resolve by @kaylarizi

File: api/src/controllers/env_info_api.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+        new_env_info = EnvInfo(
+            resource_name=data.get('resource_name'),
+            operating_system=data.get('operating_system'),
+            cpu=data.get('cpu'),
+            cpu_architecture=data.get('cpu_architecture'),
+            cpu_cores=data.get('cpu_cores'),
+            clock_speed=data.get('clock_speed'),
+            node_size=data.get('node_size')
+        )
```
[the remark this answers is not in the document]
Comment (yeudit) [reply api/src/controllers/env_info_api.py]: should resolve by @kaylarizi

File: api/src/controllers/env_info_api.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        try:
+            current_app.database_manager.add_to_db(new_env_info)
+        except Exception as e:
+            return jsonify({'error': str(e)}), 500
```
[the remark this answers is not in the document]
Comment (yeudit) [reply api/src/controllers/env_info_api.py]: should resolve by @kaylarizi

File: api/src/models/env_info.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
+from sqlalchemy import create_engine, Column, Integer, String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+# from src.models.test_suite import TestSuite
```
[the remark this answers is not in the document]
Comment (yeudit) [reply api/src/models/env_info.py:4]: should resolve by @kaylarizi

**Review by mikekcs — APPROVED** [review]

**Pull request merged**